open-policy-agent/infrastructure-general/namespace-regulation-for-pods/template.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8snamespaceregulation
+  annotations:
+    description: Disallows pods that request to run in specific namespaces, the list of disallowed namespaces provided as a parameter array from the constraint
+spec:
+  crd:
+    spec:
+      names:
+        kind: k8sNamespaceRegulation
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          properties:
+            namespaces:
+              type: array
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8snamespaceregulation
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "Pod"
+          input.parameters.namespaces[i] == input.review.object.metadata.namespace
```

Comment:
I think that it'd be best to validate the namespaces via the constraint itself, since you're able to make this check out-of-the-box.

The constraint can look like - 

```
apiVersion: constraints.gatekeeper.sh/v1beta1
kind: k8sNamespaceRegulation
metadata:
  name: pod-namespace-regulation
spec:
  enforcementAction: dryrun
  match:
    kinds:
      - apiGroups: [""]
        kinds: ["Pod"]
    namespaces:
      - default
```

The Constraint Template will look like - 
```
apiVersion: templates.gatekeeper.sh/v1beta1
kind: ConstraintTemplate
metadata:
  name: k8snamespaceregulation
  annotations:
    description: Disallows pods that request to run in specific namespaces, the list of disallowed namespaces provided as a parameter array from the constraint
spec:
  crd:
    spec:
      names:
        kind: k8sNamespaceRegulation
      validation:
        # Schema for the `parameters` field
        openAPIV3Schema:
          properties:
            namespaces:
              type: array
              items:
                type: string
  targets:
    - target: admission.k8s.gatekeeper.sh
      rego: |
        package k8snamespaceregulation

        violation[{"msg": msg}] {
          input.review.kind.kind == "Pod"
          msg := sprintf("Pod '%v' can't be created in '%v' namespace", [input.review.object.metadata.name, input.review.object.metadata.namespace])
        }
```